### DISCUSSION TITLE
JS Coroutine started as undispatched, why?

### DISCUSSION THREAD

**Discussion by lppedd** [ForteScarlet/kotlin-suspend-transform-compiler-plugin#36] [Q&A] [answered]
Hey there!  
I was just wondering what's the rationale for starting the coroutine as `UNDISPATCHED`, instead of keeping the default behavior.

https://github.com/ForteScarlet/kotlin-suspend-transform-compiler-plugin/blob/88b1bb5cceb1464ada973b2053773f35a0cea1af/runtime/suspend-transform-runtime/src/jsMain/kotlin/love.forte.plugin.suspendtrans/runtime/RunInSuspendJs.kt#L20

**Comment by ForteScarlet** [comment, marked as answer]
Hello!
To be honest, I can't quite remember exactly why I wrote that. The "me now" guesses that the "me then" was probably because I read `Deferred<T>.asPromise()`: 

https://github.com/Kotlin/kotlinx.coroutines/blob/f227432328afe41505a62a9fa6ee9d12ae6aab80/kotlinx-coroutines-core/js/src/Promise.kt#L50-L57

used `UNDISPATCHED`, so I followed suit.

Probably no deep reason 😂

**Reply by lppedd** [reply]
This is the commit that added `UNDISPATCHED`
https://github.com/Kotlin/kotlinx.coroutines/commit/03a266a7e7cf51668d51686277e7ed605f346489

Beware however that it's for converting a Promise to a Deferred, and not the other way, so I'm not sure the same "logical rules" apply.
That's interesting tho, I guess in the JS world it doesn't change much? We only have a single thread.

**Reply by lppedd** [reply]
This is the release notes for the `UNDISPATCHED` change.
https://github.com/Kotlin/kotlinx.coroutines/releases/tag/0.21.1

**Reply by ForteScarlet** [reply]
> https://github.com/Kotlin/kotlinx.coroutines/releases/tag/0.21.1

It seems to mention "'Unhandled promise rejection' warning", maybe it has something to do with something in node, but I don't know node very well so I'm not too sure about more specific details
